src/index.ts

```diff
@@ -222,6 +285,120 @@ async function getPortalItemInfo(
 
     return portalInfo;
 }
+/**
+ * Gets the set of parameters required for a given print template.
+ * @param itemId The portal item ID of the report item.
+ * @param portalUrl The URL of the ArcGIS Portal instance to use. Defaults to ArcGIS Online: "https://www.arcgis.com".
+ * @param token An optional ArcGIS token for accessing a secured report.
+ * If the report is secured, or accesses secured ArcGIS content the token is required.
+ */
+export async function getItemMetadata(
+    itemId: string,
+    portalUrl: string,
+    token?: string
+): Promise<PrintMetadata> {
+    // Fetch the portal item
+    const portalItemInfo = await getPortalItemInfo(itemId, portalUrl, token);
+
+    // Infer the URL to the reporting service from the item
+    const apiServiceUrl = `${portalItemInfo.url as unknown as string}service`;
+
+    const bearerToken = await getBearerToken(apiServiceUrl, portalUrl, token);
```

Comment:
If the plan is to use these functions from the ExB widget we might want to consider exposing them in a way that avoids having duplicate calls to get tokens and portal info.

Reply:
Refactored the getMetadata function to require a bearertoken and serviceUrl param instead of fetching them.  Also refactored the run function to accept those as run options.  If they are present it doesn't go and get them.  This should preserve backwards compatibility.